src/main/java/thePackmaster/cards/Rummage.java

```diff
@@ -28,19 +28,29 @@ public void use(AbstractPlayer p, AbstractMonster m) {
             public void update() {
                 isDone = true;
                 ArrayList<AbstractCard> possCardsList = new ArrayList<>();
-
-                for (AbstractCard q : AbstractDungeon.player.hand.group) {
-                    if (Wiz.getLogicalCardCost(q) > 0)
-                        possCardsList.add(q);
-                }
+                ArrayList<AbstractCard> chosenCards = new ArrayList<>();
+                int highestcost = 0;
 
                 for (int i = 0; i < magicNumber; i++) {
+                    for (AbstractCard q : AbstractDungeon.player.hand.group) {
+                        if (!chosenCards.contains(q)) {
```

Comment:
There's a bug here with X-cost cards. Cost reduction is meaningless for X-cost cards, but nothing in this code filters them out, and `Wiz.getLogicalCardCost` returns your current energy total for an X-cost card.

The fix is to filter out those cards on this line:
```suggestion
                        if (!chosenCards.contains(q) && q.cost > 0) {
```

Checking `q.cost > 0` will exclude three categories of cards that this code should avoid: unplayable cards, X-cost cards, and 0-cost cards. In all those cases the cost reduction is meaningless but Rummage will still act like its reducing their cost.